### ISSUE TITLE
No error for using undefined variable if previously referenced in unrelated `empty()` check

### ISSUE THREAD

**Issue by php4fan** [DEVSENSE/phptools-docs#681]
In this code:

![image](https://github.com/user-attachments/assets/a72b9192-8159-4eff-becf-22430d47179b)

I understand why the first and second references to `$avar` are not highlighted as errors, but the last one should be.

The fact that I've called `empty()` on it previously is irrelevant. You can't just assume that because I thought about it once I'm not forgetting to do the same again.

Even more absurd:

![image](https://github.com/user-attachments/assets/2546f1eb-fbd1-4353-9ab0-c48db26b0c3f)

You are correctly highlighting it as error inside the `if` block, but then, you're still failing to highlight the last occurrence.

If I remove the call to `empty()` then it's highlighted as expected:

![image](https://github.com/user-attachments/assets/03a80c17-a0d4-433b-86cd-5e35d169d1fd)

**Comment by jakubmisek** [comment]
Yes, the `empty` check overrides everything we think we know about that variable and the result is not helpful.

Regarding the error, we underlined the first occurrence only for simplicity.

**Comment by php4fan** [comment]
> Regarding the error, we underlined the first occurrence only for simplicity.

Normally that's not the case when `empty()` is not interfering.

Example 1:

![image](https://github.com/user-attachments/assets/666f12c4-2e6f-4cf5-bb51-7463e3ffbee0)

Example 2:

![image](https://github.com/user-attachments/assets/8bec1d47-f150-46ff-80f8-2826ce1a5e0a)


And these are basically like the examples in the original report:

![image](https://github.com/user-attachments/assets/daae9076-0bb7-447b-b612-78901cf77487)

![image](https://github.com/user-attachments/assets/c141a867-b1ec-412c-a078-ce53bde7f484)
